src/sources/vector/v2.rs

```diff
@@ -163,6 +162,25 @@ async fn run(
     let stream = listener.accept_stream().map(|result| {
         result.map(|socket| {
             let peer_addr = socket.connect_info().remote_addr;
+            // TODO: Primary downside to this approach is that it's counting the raw bytes on the wire, which
+            // will almost certainly be a much smaller number than the actual number of bytes when
+            // accounting for compression.
```

Comment:
This warrants an issue, if there isn't one already.

Reply:
Agreed. Meant to do so after chatting with Jesse about it before, but forgot. 🤦🏻 

Rectified: #12574